### PR TITLE
[IMP] account*,l10n*: add named i18n parameters and escape strings

### PR DESCRIPTION
**Pull request by sovi-odoo**
- Name translated string parameters to make it possible to move them around in translations and make the translators' job a bit nicer
- Make sure all rendered strings go through Markup or html_escape

In general, if a named parameter isn't useful to translators then it won't be added here (e.g. strings that already provide enough context with one parameter)

task-3884350

**Enterprise PR:** odoo/enterprise#62348

### CODE REVIEW

File: addons/account_edi_ubl_cii/models/account_edi_common.py [hunk 1]
```diff
@@ -244,15 +244,15 @@ def _check_required_fields(self, record, field_names, custom_warning_message="")
...
         # field is not present
         if custom_warning_message or isinstance(record, dict):
-            return custom_warning_message or _("The element %s is required on %s.", record, ', '.join(field_names))
+            return custom_warning_message or _("The element %(element)s is required on %(name)s.", element=record, name=', '.join(field_names))
 
         display_field_names = record.fields_get(field_names)
         if len(field_names) == 1:
             display_field = f"'{display_field_names[field_names[0]]['string']}'"
-            return _("The field %s is required on %s.", display_field, record.display_name)
+            return _("The element %(element)s is required on %(name)s.", element=display_field, name=record.display_name)
```
Comment (malb-odoo) [inline addons/account_edi_ubl_cii/models/account_edi_common.py:252]: let's keep "The field %(element), let's not changed what the user will see :smile:

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def _l10n_in_edi_ewaybill_cancel_invoice(self, invoices):
             if "312" in error_codes:
                 # E-waybill is already canceled
                 # this happens when timeout from the Government portal but IRN is generated
-                error_message = "<br/>".join(["[%s] %s" % (e.get("code"), html_escape(e.get("message"))) for e in error])
+                error_message = "<br/>".join([html_escape("[%s] %s" % (e.get("code"), e.get("message"))) for e in error])
```
Comment (malb-odoo) [inline addons/l10n_in_edi_ewaybill/models/account_edi_format.py:156]: Dont we want to also have explicit names ? same for the other htlm escape :smile:

**Review by malb-odoo — COMMENTED** [review]
Small comments :smile: Good job ! Also not sure it's a REF, i would say it's more of an IMP

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py (hunk 2 above)
Comment (sovi-odoo) [reply addons/l10n_in_edi_ewaybill/models/account_edi_format.py:156]: I don't think so. This is not getting translated and isn't error prone either so I'd keep it like that (imo)

**Review by malb-odoo — APPROVED** [review]
LGTM Thanks ! oops there is a conflict to deal with :grimacing:

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**